cli/azd/pkg/tools/azcli/function_app.go

```diff
@@ -0,0 +1,61 @@
+package azcli
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/convert"
+)
+
+type AzCliFunctionAppProperties struct {
+	HostNames []string
+}
+
+func (cli *azCli) GetFunctionAppProperties(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroup string,
+	appName string,
+) (*AzCliFunctionAppProperties, error) {
+	client, err := cli.createWebAppsClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	webApp, err := client.Get(ctx, resourceGroup, appName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving webapp properties: %w", err)
+	}
+
+	return &AzCliFunctionAppProperties{
+		HostNames: []string{*webApp.Properties.DefaultHostName},
+	}, nil
+}
+
+func (cli *azCli) DeployFunctionAppUsingZipFile(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroup string,
+	appName string,
+	deployZipPath string,
```

Comment:
A refactoring for us to consider in the future is to change this to be an `io.Reader`.  Long term we could perhaps avoid spilling these zip files to disk.